### PR TITLE
deleted admin model, working version. But admin checks can be cleaner

### PR DESCRIPTION
**Pull request by tommywei110**
it's a lot easier than expected

### CODE REVIEW

**Comment by longhtong** [comment]
Please add testadminuser@berkeley.edu to ADMIN_EMAILS to your env file or application.yml file

File: features/support/setup.rb [hunk 1]
```diff
@@ -68,7 +68,7 @@
 
 # TODO: Make this a generic email when you don't need to store
 # admin emails in the app secrets
-OmniAuth.config.mock_auth[:google_oauth2] = OmniAuth::AuthHash.new({
+OmniAuth.config.mock_auth[:google_oauth2_admin] = OmniAuth::AuthHash.new({
```
Comment (tommywei110) [inline features/support/setup.rb]: this is not the right syntax. i'll change this

File: features/step_definitions/admin_steps.rb [hunk 2]
```diff
@@ -6,8 +6,31 @@
 
 # A wrapper around the omniauth link.
 # TODO: This should be adapted to take in a provider.
-Then /I can log in with Google/ do
+Then /I can log in with Google as Admin/ do
   allow(Teacher).to receive(:validate_auth).and_return(true)
-  Rails.application.env_config["omniauth.auth"] = OmniAuth.config.mock_auth[:google_oauth2]
+  Rails.application.env_config["omniauth.auth"] = OmniAuth.config.mock_auth[:google_oauth2_admin]
```
Comment (tommywei110) [inline features/step_definitions/admin_steps.rb]: this isn't quite used like this

**Review by longhtong — APPROVED** [review]

**Review by justinphyo — APPROVED** [review]
Tests are passing locally and site is working as expected.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -7,9 +7,12 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
 
-  # TODO: This really need to be tweaked!
   def is_admin?
-    @is_admin ||= session.key?("logged_in") and session[:logged_in] == true
+    @is_admin = logged_in? ? current_user.admin : false
```
Comment (cycomachead) [inline app/controllers/application_controller.rb:11]: This is good. Minor note  but you can use `.admin?` when a column is a boolean value, which is a little more clear.

File: db/schema.rb [hunk 4]
```diff
@@ -10,24 +10,12 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2021_02_16_010033) do
+ActiveRecord::Schema.define(version: 2021_02_19_233254) do
+# ActiveRecord::Schema.define(version: 2021_02_16_010033) do
```
Comment (cycomachead) [inline db/schema.rb:14]: You can just delete this line in the future. :)

**Pull request merged**